rfcs/sliders-component.md

```diff
@@ -0,0 +1,122 @@
+# Sliders Component
+
+Represents control with sliding axis to choose a single value or range between min and max values.
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Sliders} from 'baseui/sliders';
+
+export default () => <Sliders range={[
+  {value: 0, label: '0, min'},
+  {value: 100, label: '100, max'}
+]} />;
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {Foo} from 'baseui/foo';
+
+export default () => {
+  return <Sliders
+    range={[
+      {value: 0, label: '0, min'},
+      {value: 20, label: '20'},
+      {value: 40, label: '40'},
+      {value: 60, label: '60'},
+      {value: 80, label: '80'},
+      {value: 100, label: '100, max'}
+    ]}
+    thumbs={4}
+    initialState={{
+      selectedValues: [
+        {value: 20, label: '20'}, {value: 40, label: '40'},
+        {value: 60, label: '60'}, {value: 80, label: '80'},
+      ]
+    }}
+    step={20}
+    onChange={(event, selectedValues)=>{console.log(selectedValues)}}
+    overrides={{
+      Thumb: props => <div style={{
+        color: 'red',
+        borderRadius: '5px',
+      }}></div>,
+    }}
+    />
+}
+```
+
+## Exports
+
+* `Sliders`
+* `StatefulSliders`
+* `StatefulSlidersContainer`
+* `StyledRoot`
+* `StyledAxis`
+* `StyledTick`
+* `StyledThumb`
+* `CONSTANT`
+
+## `Sliders` API
+
+* `range: {Array<Point>}` - Required
+  All points present on slider axis. First and last point represent min and max values. Each `Point` can be a value itself (for primitive type) or if `Point` is `Object` it should have `value` and `label`. If more than 2 elements are present in array, they represent all ticks on axis and `step` property is ignored.
+* `thumbs: number` - Optional
+  Amount of thumbs representing how many values can be selected. For even quantity of thumbs selected range is considered in between them. Default is `1`
```

Comment:
Seems like supporting more than 2 thumbs might complicate things quite a bit. I'm trying to figure out how a slider with 4 or 5 thumbs would work. You'd either need to have a bunch of logic/styles to deal with overlapping ranges, or restricting the min of one range to the max of the other. 

What do you think about just simplifying to either 1 or 2 thumbs?

Reply:
I'd go for a simplified version too as a first approach - if later needed, we can figure it out then.

Reply:
Yes, more than 2 seems to be very rare case, for instance cut couple samples on timeline or signal axis. However, I see the ways to do it without complicating logic to support it. As the benefit, the interface would not care about how many thumbs there, all unified. No need to check for "magic numbers" 1 or 2

Reply:
Maybe we can discuss this at weekly meeting on monday?

Reply:
ok

Reply:
Let's do a more widely used case (two thumbs) for the initial implementation and then we can look into the improvements once we have a design and use case for it.

Reply:
We can do `multiple=true` to mean you want a slider with 2 knobs. Anything more than 2 knobs sounds like a really obscure use case to me. The `<select>` in [HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select) also has `multiple` attribute, I'm not sure if we should call it `multiple` because of this.

`react-select` calls this `isMulti`.

Reply:
Cool, thanks for the insights everyone. Sounds like we should simplify to just 1 or 2 (both pretty common use cases).

I'm leaning towards just implicitly switching to two thumbs if `value` is an array (instead of a single value), rather than having an explicit `thumbs` prop. Curious if anyone else has thoughts here.

Reply:
Thanks for the insights all.

I like the idea of having a boolean prop like kahwee mentioned instead of `thumbs: number`. Antd uses `range: boolean`, but we're already using `range` for something else, so maybe `multiple` is good? In theory we could also implicitly detect this if `value` is an array.

Reply:
I added explicit `thumbs` as initial value of selected range was optional, so if user doesn't provide it, we can't say, how many thumbs needed. I can do multiple or make initial value required as it should be more required than optional. otherwise will have to make knobs min amd max, which looks wierd

Reply:
so removed `thumb` property and made `value` as required, in this case it can be 1 or 2 point value indicating what to expect from component